src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/repository/TaskResourceRepository.java

```diff
@@ -77,6 +79,14 @@ List<TaskResource> findByCaseIdInAndStateInAndReconfigureRequestTimeIsNull(
     List<TaskResource> findByStateInAndReconfigureRequestTimeGreaterThan(
         List<CFTTaskState> states, OffsetDateTime reconfigureRequestTime);
 
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({@QueryHint(name = LOCK_TIMEOUT_STR, value = "0")})
```

Comment:
So I think we should think about extending the lock timeout here.  The behaviour we want is to wait for any other threads to complete an update before we run our query.  I think a value of 0 would throw an exception straight away which may not be what we want e.g. if the previous thread is a Task Assignment rather than a Cancellation.  I think 5000 millis is enough to start with.